server/src/main/java/io/seata/server/storage/db/store/LogStoreDataBaseDAO.java

```diff
@@ -334,16 +334,26 @@ public boolean insertBranchTransactionDO(BranchTransactionDO branchTransactionDO
 
     @Override
     public boolean updateBranchTransactionDO(BranchTransactionDO branchTransactionDO) {
-        String sql = LogStoreSqlsFactory.getLogStoreSqls(dbType).getUpdateBranchTransactionStatusSQL(branchTable);
+        boolean updateAppData = StringUtils.isNotBlank(branchTransactionDO.getApplicationData());
+        String sql = updateAppData ?
+                LogStoreSqlsFactory.getLogStoreSqls(dbType).getUpdateBranchTransactionAppDataSQL(branchTable)
+                : LogStoreSqlsFactory.getLogStoreSqls(dbType).getUpdateBranchTransactionStatusSQL(branchTable);
         Connection conn = null;
         PreparedStatement ps = null;
         try {
             conn = logStoreDataSource.getConnection();
             conn.setAutoCommit(true);
             ps = conn.prepareStatement(sql);
             ps.setInt(1, branchTransactionDO.getStatus());
-            ps.setString(2, branchTransactionDO.getXid());
-            ps.setLong(3, branchTransactionDO.getBranchId());
+            if (updateAppData) {
+                ps.setString(2, branchTransactionDO.getApplicationData());
+                ps.setString(3, branchTransactionDO.getXid());
+                ps.setLong(4, branchTransactionDO.getBranchId());
+            } else {
+                ps.setString(2, branchTransactionDO.getXid());
+                ps.setLong(3, branchTransactionDO.getBranchId());
+            }
+
```

Comment:
**more elegant code:** 
```
int i = 1;
ps.setXxxx(i++, xxxx);
if (updateAppData) {
    ps.setXxxx(i++, xxxx);
}
ps.setXxxx(i++, xxxx);
ps.setXxxx(i++, xxxx);
```

Reply:
done & thanks

Reply:
Please simplify the code.  That's why I suggest you use `i++`
```
  if (updateAppData) {
    ps.setString(i++, branchTransactionDO.getApplicationData());
  }
  ps.setString(i++, branchTransactionDO.getXid());
  ps.setLong(i++, branchTransactionDO.getBranchId());
```